pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentDispatcherSingleActiveConsumer.java

```diff
@@ -146,4 +147,9 @@ protected void readMoreEntries(Consumer consumer) {
     protected void cancelPendingRead() {
         // No-op
     }
+
+    public CompletableFuture<Void> close() {
+        IS_CLOSED_UPDATER.set(this, TRUE);
+        return disconnectAllConsumers();
```

Comment:
Since this part is still common, we could keep in the abstract class and then call `super.close()` where needed

Reply:
Restored `AbstractDispatcherSingleActiveConsumer#close()`. Only `PersistentDispatcherSingleActiveConsumer` overrides it.